### PR TITLE
feature tested

### PR DESCRIPTION
**Pull request by Mithi-code**
### Testing

The following changes were made in this milestone:

🖌️  Snapshots were created for each component
🖌️  Jest and React-testing-library was used to test the  code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 > Math magicians" is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to: Make simple calculations. Read a random math-related quote.
 
 ![screenshot](./public/app1.png)
-
+![test's screenshot](./public/test.png)
```
Comment (D4nielJ) [inline README.md:8]: - [ ] Please remove the default test that comes with CRA when the app was first created, as it's currently failing.

image

File: src/__test__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import calculate from '../logic/calculate';
+import obj from '../__mock__/calculate';
+
+describe('Calculate', () => {
+  test('the test is to return an empty object', () => {
+    expect(calculate(obj, 'AC')).toEqual(obj);
+    expect(calculate(obj, 'AC')).toMatchSnapshot();
```
Comment (D4nielJ) [inline src/__test__/calculate.test.js:7]: - [ ] I think we're using incorrectly the snapshot functionality here. Just checking that calculate returns the expected result, in this case, is enough. Instead, we want to use snapshots to make "copies" of a react component that should rerender the same every time. Let's take a look at this example:

```
import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../Link.react';

it('renders correctly', () => {
  const tree = renderer
    .create(<Link page="http://www.facebook.com">Facebook</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
```
In this case a snapshot of the Link component it's being created and in the future, it should always match the snapshot unless a change to the Link component is made.

- [ ] Please write snapshots tests for your components: Home, Calculator, Quote, etc. [You can read more about how to take snapshots here.](https://jestjs.io/docs/snapshot-testing)

File: README.md (hunk 1 above)
Comment (D4nielJ) [reply README.md:8]: - [ ] Please write tests that simulate user interaction as required by Microverse.

> - Create unit tests for all React components:
>   - Use Jest snapshots to test the components.
>   - Use React Testing Library to simulate user interaction.

This should simulate clicks on links, clicks on the calculator, and expect the React Tree to rerender properly every time. These are especially useful as they allow to test the integration of the logic and the UI much quicker than testing the app manually each time. [Read more about it here.](https://testing-library.com/docs/react-testing-library/example-intro)

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @Mithi-code and @fobadara, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: 
- Unit tests for the `calculate` and `operate` functions have been written! :rocket: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
## STATUS: APPROVED :tada: :1st_place_medal: :tada: 
Hi :wave: @Mithi-code , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**